pkg/schema/annotations.go

```diff
@@ -20,36 +19,35 @@ const (
 )
 
 type Annotation interface {
-	NewTypeFromAnn() yamlmeta.Type
+	NewTypeFromAnn() (yamlmeta.Type, error)
 }
 
 type TypeAnnotation struct {
-	any          bool
-	inferredType yamlmeta.Type
-	itemPosition *filepos.Position
+	any  bool
+	item yamlmeta.Node
 }
 
 type NullableAnnotation struct {
-	providedValueType yamlmeta.Type
-	itemPosition      *filepos.Position
+	item yamlmeta.Node
 }
 
 // DefaultAnnotation is a wrapper for a value provided via @schema/default annotation
 type DefaultAnnotation struct {
 	val interface{}
```

Comment:
🤔 💭  Curious that `TypeAnnotation` and `NullableAnnotation` both have a reference to their host `yamlmeta.Node` but `DefaultAnnotation` does not...